### ISSUE TITLE
SQLite API: Make Row Reader more efficient and backward compatible

### ISSUE THREAD

**Issue by fklebert** [ndsev/zserio#566]
At the moment the zserio generated API for Rows of a SQLite table does kind of a "SELECT *" from the relevant table. It does not really use "*", but all the column names that are available in the schema.
This is in two cases not optimal:
1. It makes SQLite fetch all columns, which in many cases is not necessary. In case of large blobs in columns, it will generate unnessecary memory pressure and performance degradation.
2. It breaks backward compatibility with older databases in case columns have been added in subsequent versions.

It would be beneficial if we can extend `::createReader(condition)` to also allow to specify the colunms to be fetched. Like
`::createReader(columns, condition)`

**Comment by mikir** [comment]
Yes, I agree. This is a good enhancement of SQLite API. The same problem is in Java and Python API(s).
